### PR TITLE
implement registration and authentication (without css and tests)

### PR DESCRIPTION
**Pull request by SimonLab**
ref #13
This PR add the registration and login functionality to the application
It also add more tests for the different endpoints of the app, I will continue to add some more on the next PRs
The UI is implemented on this PR: #51

### CODE REVIEW

File: coveralls.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "coverage_options": {
-    "minimum_coverage": 40
+    "minimum_coverage": 10
```
Comment (shouston3) [inline coveralls.json]: lol, fine for now 👍

File: web/templates/search/index.html.eex [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="jumbotron">
+  <h2>BestEvidence</h2>
+  <%= form_for @conn, search_path(@conn, :new), [as: :search, method: :get], fn f -> %>
+    <div class="input-group">
+      <%= text_input f, :search, placeholder: "Search", class: "form-control" %>
+      <span class="input-group-btn">
+        <%= submit "Search", class: "btn btn-default" %>
+      </span>
+    </div>
+  <%= end %>
```
Comment (shouston3) [inline web/templates/search/index.html.eex]: `<%= end %>` should be `<%= end %>`

File: coveralls.json (hunk 1 above)
Comment (SimonLab) [inline coveralls.json]: yes it will go up very soon

**Comment by SimonLab** [comment]
@shouston3 I will create an issue to see if we can use Guardian, I think I need a bit more time to investigate it

**Comment by shouston3** [comment]
If your looking to investigate it, I would recommend going though that tutorial I linked, it takes about half an hour to get through and will give you a better auth configuring (handling browser session cookies, better security, etc)

**Comment by shouston3** [comment]
This initial auth should be fine for now

**Pull request merged**